File: GallerySample/app/src/main/java/com/belatrix/fundamentals/MainActivity.java
package com.belatrix.fundamentals;

import android.animation.Animator;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.ViewAnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private ImageView imageViewBx;
    private TextView textViewHello;
    private View constraintLayoutContainer;
    private Snackbar snackbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ui();
        app();
    }

    private void app() {
        imageViewBx.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //bla bla bla bla
                //updateText();
                rippleEffect();
                //showMessage();
            }
        });
    }

    private void showMessage(){
        snackbar.setText("Hello Belatrix");
        snackbar.show();
    }
    private void rippleEffect() {

        int cx = (imageViewBx.getLeft() + imageViewBx.getRight()) / 2;
        int cy = (imageViewBx.getTop() + imageViewBx.getBottom()) / 2;

        int finalRadius = Math.max(constraintLayoutContainer.getWidth()/2, constraintLayoutContainer.getHeight()/2);

        Animator anim =
                null;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
            anim = ViewAnimationUtils.createCircularReveal(constraintLayoutContainer, cx, cy, 0, finalRadius);
            anim.setDuration(2000);
            anim.start();
        }
    }

    private void updateText() {
        textViewHello.setText("Hello Android");
    }

    private void ui() {
        constraintLayoutContainer= findViewById(R.id.constraintLayoutContainer);
        imageViewBx= (ImageView) findViewById(R.id.imageViewBx);
        textViewHello= (TextView) findViewById(R.id.textViewHello);

        snackbar= Snackbar.make(constraintLayoutContainer,"Message",Snackbar.LENGTH_LONG);
    }
}

File: training/Exercises/app/src/main/java/com/belatrix/kotlintemplate/MainActivity.kt
package com.belatrix.kotlintemplate

import android.animation.Animator
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.view.ViewAnimationUtils
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var snackbar:Snackbar?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ui()
        app()
    }

    fun ui(){
        snackbar= Snackbar.make(constraintLayoutContainer,"Message",Snackbar.LENGTH_LONG)
    }

    fun app(){
        imageViewBx.setOnClickListener({
            //bla bla bla bla
            //updateText()
            //showMessage()
            rippleEffect()
        })
    }

    fun updateText(){
        textViewHello.text="Hello Android"
    }

    fun showMessage(){
        snackbar!!.setText("Hello Belatrix")
        snackbar!!.show()
    }

    fun rippleEffect(){
        val cx:Int= (imageViewBx.left+imageViewBx.right)/2
        val cy:Int= (imageViewBx.top+imageViewBx.bottom)/2

        val finalRadius= Math.max(constraintLayoutContainer.width/2, constraintLayoutContainer.height/2)

        var anim:Animator?=null;

        if(android.os.Build.VERSION.SDK_INT>=android.os.Build.VERSION_CODES.LOLLIPOP){
            anim= ViewAnimationUtils.createCircularReveal(constraintLayoutContainer,cx,cy,0.toFloat(),finalRadius.toFloat())
            anim.duration=2000
            anim.start()
        }
    }
}
